internal/olm/client.go

```diff
@@ -249,135 +232,31 @@ func (c Client) doRequest(ctx context.Context, url string) (*http.Response, erro
 	return resp, nil
 }
 
-func decodeResources(rds ...io.Reader) ([]unstructured.Unstructured, error) {
-	var objs []unstructured.Unstructured
+func toObjects(us ...*unstructured.Unstructured) (objs []runtime.Object) {
+	for _, u := range us {
+		objs = append(objs, u)
+	}
+	return objs
+}
+
+func decodeResources(rds ...io.Reader) (objs []*unstructured.Unstructured, err error) {
```

Comment:
Any reason to change this from `[]unstructured.Unstructured` to `[]*unstructured.Unstructured`, which causes all the cascading changes in other function signatures?

If it's for the `toObjects()` function that requires a pointer to satisfy the `runtime.Object` interface, could we update that to:
```go
func toObjects(us ...unstructured.Unstructured) (objs []runtime.Object) {
	for _, u := range us {
		objs = append(objs, &u)
	}
	return objs
}
```

Reply:
I don't mind changing pointers back to values; this change is part of an original draft that I ended up not going with.